packages/webvh/src/cryptosuites/eddsa-jcs-2022.ts

```diff
@@ -0,0 +1,141 @@
+import { Buffer } from 'buffer'
+import { createPublicKey, verify } from 'crypto'
+import { type AgentContext, type VerificationMethod } from '@credo-ts/core'
+import { DidsApi, MultiBaseEncoder, Sha256 } from '@credo-ts/core'
+import { canonicalize } from 'json-canonicalize'
+import { WebVhResource } from '../anoncreds/utils/transform'
+import { ProofOptions } from './types'
+
+export class EddsaJcs2022Cryptosuite {
+  agentContext: AgentContext
+  proofOptions: object = {
+    type: 'DataIntegrityProof',
+    cryptosuite: 'eddsa-jcs-2022',
+  }
+  constructor(agentContext: AgentContext) {
+    this.agentContext = agentContext
+  }
+
+  public async _logError(error: string) {
+    this.agentContext.config.logger.error(error)
+  }
+
+  public async _publicBytesFromVerificationMethodId(verificationMethodId: string): Promise<Uint8Array | undefined> {
+    const didsApi = this.agentContext.dependencyManager.resolve(DidsApi)
+    const didDocument = await didsApi.resolveDidDocument(verificationMethodId as string)
+
+    let verificationMethod: VerificationMethod | undefined
+    if ((verificationMethodId as string).includes('#')) {
+      const fragment = (verificationMethodId as string).split('#')[1]
+      verificationMethod = didDocument.verificationMethod?.find((vm: VerificationMethod) =>
+        vm.id.endsWith(`#${fragment}`)
+      )
+    }
+
+    if (!verificationMethod) {
+      this._logError('Could not find verification method in did:webvh DID document')
+      return
+    }
+
+    if ('publicKeyMultibase' in verificationMethod && verificationMethod.publicKeyMultibase) {
+      const publicKeyBytes = this._publicKeyBytesFromMultikey(verificationMethod.publicKeyMultibase)
+      return publicKeyBytes
+    }
+    this._logError('Could not find verification method in did:webvh DID document')
+    return
+  }
+
+  public _publicKeyBytesFromMultikey(multikey: string) {
+    const publicMultikeyBytes = MultiBaseEncoder.decode(multikey).data
+    const publicMultikeyHex = Array.from(publicMultikeyBytes)
+      .map((n) => n.toString(16).padStart(2, '0'))
+      .join('')
+    const publicKeyHex = publicMultikeyHex.substring(4)
+    const publicKeyLength = publicKeyHex.length / 2
+    const publicKeyBytes = new Uint8Array(publicKeyLength)
+    for (let i = 0; i < publicKeyLength; i++) {
+      publicKeyBytes[i] = Number.parseInt(publicKeyHex.substr(i * 2, 2), 16)
+    }
+    return publicKeyBytes
+  }
+
+  public _keyFromPublicBytes(publicKeyBytes: Uint8Array) {
+    const publicKey = createPublicKey({
+      key: Buffer.concat([
+        Buffer.from([0x30, 0x2a, 0x30, 0x05, 0x06, 0x03, 0x2b, 0x65, 0x70, 0x03, 0x21, 0x00]),
+        Buffer.from(publicKeyBytes),
+      ]),
+      format: 'der',
+      type: 'spki',
+    })
+    return publicKey
+  }
```

Comment:
You can use `getPublicJwkFromVerificationMethod` to get the publicJwk from the verification method, which then can do all these transformations. It's better to reuse the existing utils for this, as there's a lot of transformation logic already in place

Reply:
I've implemented getPublicJwkFromVerificationMethod